starter/src/test/java/io/javaoperatorsdk/operator/springboot/starter/AutoConfigurationIntegrationTest.java

```diff
@@ -0,0 +1,100 @@
+package io.javaoperatorsdk.operator.springboot.starter;
+
+import java.io.IOException;
+import java.nio.file.Files;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.k3s.K3sContainer;
+import org.testcontainers.utility.DockerImageName;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.ListOptionsBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.javaoperatorsdk.operator.api.config.ConfigurationServiceProvider;
+import io.javaoperatorsdk.operator.springboot.starter.CrdUploader.CrdTransformer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.slf4j.LoggerFactory.getLogger;
+
+@SpringBootTest(properties = {"javaoperatorsdk.crd.apply-on-startup = true"})
+public class AutoConfigurationIntegrationTest {
+
+  @Autowired
+  private KubernetesClient kubernetesClient;
+
+  @BeforeAll
+  static void beforeAll() {
+    ConfigurationServiceProvider.reset();
+  }
+
+  @Test
+  void crdsUploadedAndTransformersApplied() {
+    var list = kubernetesClient.apiextensions().v1()
+        .customResourceDefinitions()
+        .list(new ListOptionsBuilder().withLabelSelector("Glory = Glory, Man = United").build())
+        .getItems();
+
+    var crds = assertThat(list).hasSize(2);
+    crds.filteredOn(it -> "testresource".equals(it.getSpec().getNames().getSingular()))
+        .hasSize(1)
+        .filteredOn(it -> "sample.javaoperatorsdk".equals(it.getSpec().getGroup()))
+        .hasSize(1);
+    crds.filteredOn(it -> "josdksecondresources".equals(it.getSpec().getNames().getPlural()))
+        .hasSize(1)
+        .filteredOn(it -> "josdk.io".equals(it.getSpec().getGroup()))
+        .hasSize(1);
+  }
+
+  @TestConfiguration
+  static class TestConfig {
+
+    static {
+      System.setProperty("kubernetes.namespace", "default");
+    }
+
+    @Bean(name = "k3sContainer", initMethod = "start", destroyMethod = "stop")
```

Comment:
alternative woudl be to use jenvtest: https://github.com/java-operator-sdk/jenvtest
Although there were reported there some issues, that we might take a look too in close future.

Reply:
oh nice, will have a look at this